src/main/java/technology/tabula/extractors/BatchSelectionExtractor.java

```diff
@@ -0,0 +1,430 @@
+package technology.tabula.extractors;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import technology.tabula.ObjectExtractor;
+import technology.tabula.Page;
+import technology.tabula.PageIterator;
+import technology.tabula.Rectangle;
+import technology.tabula.Table;
+import technology.tabula.detectors.NurminenDetectionAlgorithm;
+import technology.tabula.detectors.StringSearch;
+import technology.tabula.detectors.SpreadsheetDetectionAlgorithm;
+import technology.tabula.writers.CSVWriter;
+import technology.tabula.writers.Writer;
+
+// NOTES:
+//		need to remove tables from auto/spread list if they are used as a best guess
+//		or remove very similar tables from the list before extracting data
+public class BatchSelectionExtractor {
```

Comment:
It might be worthwhile for this to mimic https://github.com/tabulapdf/tabula-java/blob/master/src/main/java/technology/tabula/extractors/SpreadsheetExtractionAlgorithm.java#L85 and have the `extract()` method return a list of tables, rather than writing the output to disk directly. 

I think it would also work for it to return a list of lists of tables (one list of tables per document in the batch job) or even a dictionary/hash mapping batch document filenames to a list of tables. 

This will also, I believe, solve one of the problems that's causing the continuous integration job to fail. Rather than requiring that an output directory exist, this `extract` method can "write" the output to a stream object, rather than a real file. That way, the tests aren't side-effecty in terms of creating files and folders.

Incidentally, I think there needs to be a little bit of work to deal with Linux/Mac compatibility. Where the test for this method is supposed to write some files to the `/src/test/resources/technology/tabula/batch/output/` directory, I end up with a file in `java/src/test/resources/technology/tabula/batch/` called `output\well_text_a.csv`. Java has the ability to let you name files without using `\` or `/` and letting Java figure out the right one to use.